api/v1alpha1/tunnelbinding_types.go

```diff
@@ -118,6 +119,141 @@ type TunnelBindingStatus struct {
 	Services  []ServiceInfo `json:"services"`
 }
 
+type AccessConfig struct {
+	// Enable handling of access configuration
+	//+kubebuilder:validation:Optional
+	//+kubebuilder:default:=false
+	Enabled bool `json:"enabled"`
+	// Application type self_hosted,saas
+	//+kubebuilder:validation:Optional
+	//+kubebuilder:validation:Enum:="";"self_hosted";"saas"
+	//+kubebuilder:default:="self_hosted"
+	Type string `json:"type"`
+	// List of access policies
+	//+kubebuilder:validation:Optional
+	AccessPolicies []AccessPolicy `json:"accessPolicies"`
+	// Application settings
+	//+kubebuilder:validation:Optional
+	Settings AccessConfigSettings `json:"settings"`
+}
```

Comment:
Since this is being used as part of the tunnelBinding, can we use only the configs that makes sense?

For example, the type would always need to be self_hosted, saas would never work. Do we need the enabled flag? If the config isn't specified, it is disabled.